src/core/Akka.Cluster/ClusterDaemon.cs

```diff
@@ -1003,6 +1002,7 @@ internal static string VclockName(UniqueAddress node)
         private ImmutableList<Address> _seedNodes;
         private IActorRef _seedNodeProcess;
         private int _seedNodeProcessCounter = 0; //for unique names
+        private Deadline _joinSeedNodesDeadline;
```

Comment:
Call me old fashioned, but I'd prefer it to be a `CancellationToken` instead - this way we could also use it inside `await Cluster.JoinSeedNodesAsync(nodes, token)`

Reply:
Just like any other argument passed there - as a field in the message. This would need a shift from the canonical Scala implementation, but IMO they don't support any cancellation of calls whatsoever, while this is:

1. Pretty much obvious in .NET.
2. Allows you to specify more interesting patterns like retrying to join with widening/narrowing timeout window - ofc this can be done with current design too, but it would require recreating ActorSystem with different config parameter value each time.

Reply:
Maybe this could be improved upon a later PR.